azure/install-rust.yml

```diff
@@ -54,8 +54,9 @@ steps:
     displayName: Check installed rust version
 
   # Checkout any submodules the repository may have
-  - checkout: self
```

Comment:
As posted in the other thread, I still think it this is inappropriate coupling.  I'm not a fan of "convenience" for messing with a public API.

Reply:
I see your point. Give me a second to throw something together.

Reply:
Take a look at https://github.com/crate-ci/azure-pipelines/pull/46/commits/a1aa272a1fc8e3f8c3692817fae2e4ef6b47db79 and see what you think? Followed your `_` for private templates suggestion.